test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/DeleteTests.cs

```diff
@@ -121,6 +121,34 @@ async Task ExecuteAndValidateNotFoundStatus(Func<Task> action)
             }
         }
 
+        [HttpIntegrationFixtureArgumentSets(DataStore.CosmosDb, Format.All)]
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAResource_WhenPurging_ThenServerShouldDeleteHistoryAndKeepCurrentVersion()
+        {
+            using FhirResponse<Observation> response = await _client.CreateAsync(Samples.GetDefaultObservation().ToPoco<Observation>());
+
+            string resourceId = response.Resource.Id;
+            string historyVersion = response.Resource.Meta.VersionId;
+
+            response.Resource.Meta.Tag.Add(new Coding("tests", Guid.NewGuid().ToString()));
+
+            using FhirResponse<Observation> updateResponse = await _client.UpdateAsync(response.Resource);
+
+            // Delete the resource.
+            await _client.DeleteAsync($"{nameof(Observation)}/{resourceId}/$purge");
+
+            // Subsequent read on the resource should return Gone.
```

Comment:
This comment doesn't match the check that is happening after it. It looks like you are checking that the worked and the version id from a read matches the updated resource.